cmd/manager/main.go

```diff
@@ -212,3 +226,24 @@ func setProfiler() {
 		}()
 	}
 }
+
+func lockHandler() (lockfile.Lockfile, error) {
+	lockFilePath, ok := os.LookupEnv("NMSTATE_INSTANCE_NODE_LOCK_FILE")
+	if !ok {
+		return "", errors.New("Failed to find NMSTATE_INSTANCE_NODE_LOCK_FILE ENV var")
+	}
+	log.Info(fmt.Sprintf("Try to take exclusive lock on file: %s", lockFilePath))
+	handlerLock, err := lockfile.New(lockFilePath)
+	if err != nil {
+		return handlerLock, errors.Wrapf(err, "failed to create lockFile for %s", lockFilePath)
+	}
+	err = wait.PollImmediate(5*time.Second, 600*time.Second, func() (done bool, err error) {
```

Comment:
Why do we have a timeout here? Do you assume that during the upgrade, old operator will be gone within 10 minutes?

Reply:
@qinqon ^^

Reply:
Is not very rebust to leave this loop without timeout, we can increase the timeout if we want.